heartbeat/monitors/active/http/simple_transp.go

```diff
@@ -191,7 +178,7 @@ func (t *SimpleTransport) readResponse(
 
 	t.sigStartRead()
 
-	if requestedGzip && resp.Header.Get("Content-Encoding") == gzipEncoding {
+	if resp.Header.Get("Content-Encoding") == gzipEncoding {
 		resp.Header.Del("Content-Encoding")
```

Comment:
I'm wondering if these lines deleting content length and encoding etc. are still correct? They were part of the dead code path before. 

I guess they're a little confusing in that the decoded content is a different length, but I think we should keep the values since it's more correct.

Reply:
Do you think it'd be appropriate to keep these, and then add some field to the event to indicate the decompressed body length? That would be more of an enhancement I suppose. I am ++ on removing these lines as well.

Reply:
I removed those lines BTW

https://github.com/elastic/beats/pull/28739/commits/bb455435bdca7c948fca98f546280abaf8456c1e